test/unit/utils_test.py

```diff
@@ -0,0 +1,11 @@
+from pudl.utils import retry
+
+
+def test_retry(mocker):
+    func = mocker.MagicMock(side_effect=[RuntimeError, RuntimeError, RuntimeError, 1])
+    sleep_mock = mocker.MagicMock()
+    with mocker.patch("time.sleep", sleep_mock):
+        assert retry(func=func, retry_on=(RuntimeError,)) == 1
+
+    assert sleep_mock.call_count == 3
+    sleep_mock.assert_has_calls([mocker.call(2**x) for x in range(3)])
```

Comment:
Will `MagicMock` add a 7 second delay to our unit tests here? Doesn't seem to be the case when I run unit tests, but maybe I'm missing something. If so, we could reduce the number of `RuntimeErrror` calls by one to gain a bit of speed back here if this becomes a noticeable lag.